### PR TITLE
Replace NA with 0 for select data set columns

### PR DESCRIPTION
**Pull request by kylebaron**
# Summary

Missing values (`NA`) in an input data set are typically written to `.csv` files as `.` and interpreted by NONMEM as `0`. 

This PR adds functionality to automatically convert `NA` in mrgsolve input data to `0` _for select columns_ where `NA` would naturally get filled by `0`. 

The following columns will be targets for replacing `NA`:
- `AMT` 
-  `CMT`  
- `EVID` 
- `II`  
- `ADDL` 
- `RATE`
- `SS` 

Lower case versions are also converted. 

These names are contained in `GLOBALS$TRAN_FILL_NA`

If a parameter data set column contains `NA` we continue to generate an error. This behavior will continue to be different from NONMEM (where _every_ `NA` is converted to `0`).

Note that `check_data_set_na` now returns logical rather than `NULL`; we use this value to figure out if we need to do this processing (fill NA with 0). 

While working in `mrgindata.R`, converted `any(duplicated())` to `anyDuplicated()`; couldn't help it.

### CODE REVIEW

**Review by kyleam — APPROVED** [review]
Thanks for the detailed PR description.  Left a few minor comments, but looks good to me aside from what (I think) is a missing test assertion.

File: R/mrgindata.R [hunk 1]
```diff
@@ -252,44 +256,50 @@ valid_data_set.matrix <- function(x,verbose=FALSE) {
...
     err <- TRUE
   }  
   if(err) stop("Found missing values in input data.", call.=FALSE)
-  return(invisible(NULL))
+  return(invisible(TRUE))
+}
+
+#' Look for TRAN columns replace NA with 0
+#' Columns to scan are found in `GLOBALS$TRAN_FILL_NA`
+#' @keywords internal
```
[the remark this answers is not in the document]
Comment (kylebaron) [reply R/mrgindata.R:283]: I removed it.

File: R/mrgindata.R [hunk 2]
```diff
@@ -252,44 +256,50 @@ valid_data_set.matrix <- function(x,verbose=FALSE) {
...
+#' @noRd
+fill_tran_na <- function(data) {
+  cols_to_zero <- check_column_na(data, GLOBALS[["TRAN_FILL_NA"]])
+  for(col in cols_to_zero) {
+    to_zero <- is.na(data[,col])
+    data[to_zero, col] <- 0
+  }
```
[the remark this answers is not in the document]
Comment (kylebaron) [reply R/mrgindata.R:290]: I messed around with this sort of subsetting and selecting when I was writing the code but was getting errors that I couldn't figure out. Mine didn't work but yours dose; I'll go with it. Thanks!

File: tests/testthat/test-mrgindata.R [hunk 3]
```diff
@@ -194,3 +194,26 @@ test_that("integer64 columns are dropped from idata_set  [SLV-TEST-0012]", {
...
+
+  flagged <- mrgsolve:::check_column_na(tmp, c("ss", "aa", "II", "SS"))
+  expect_equal(flagged, "ss")
+
+  tst <- valid_data_set(tmp, house())
+  expect_equal(tst[, "ss"], c(1, 0))
+    
+  tmp2 <- tmp
+  tmp2$FOO <- NA_real_
+  flagged <- mrgsolve:::check_column_na(tmp2, mrgsolve:::GLOBALS$TRAN_FILL_NA)
```
[the remark this answers is not in the document]
Comment (kylebaron) [reply tests/testthat/test-mrgindata.R:211]: Yes. It is checked now.

**Pull request merged**